lib/features/home/ui/widgets/build_categories_section.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mentorship/core/di/dependency_injection.dart';
+import 'package:mentorship/features/home/logic/home_cubit.dart';
+import 'package:mentorship/features/home/logic/home_state.dart';
+import 'package:mentorship/features/home/ui/widgets/categories_list.dart';
+import 'package:mentorship/features/home/ui/widgets/shimmer_categories_list.dart';
+
+class BuildCategoriesList extends StatelessWidget {
+  const BuildCategoriesList({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final homeCubit = getIt<HomeCubit>();
+
+    return BlocBuilder<HomeCubit, HomeState>(
+      buildWhen: (previous, current) =>
+          current is HomeCategoriesSuccess ||
+          current is HomeCategoriesError ||
+          current is HomeCategoriesLoading,
+      builder: (context, state) {
```

Comment:
if you are already using freezed, consider using it in the state class of the cubit, to make this code better instead of the if/elses.

I mean to use when or  whenOrNull like you did in the cubit